### DISCUSSION TITLE
code questions on remix, "_currentIndex" and "_ownerships"

### DISCUSSION THREAD

**Discussion by yzh617** [hashlips-lab/nft-erc721-collection#303] [Q&A]
Hi

I paste the YourNftToken.sol on the remix, and there is an error code on compile.
the variable "_currentIndex" and "_ownerships", I did not find any declaration for the variable, 
some one know why?  thanks for the big help.

    function walletOfOwner(address _owner) public view returns (uint256[] memory) {
    uint256 ownerTokenCount = balanceOf(_owner);
    uint256[] memory ownedTokenIds = new uint256[](ownerTokenCount);
    uint256 currentTokenId = _startTokenId();
    uint256 ownedTokenIndex = 0;
    address latestOwnerAddress;

    while (ownedTokenIndex < ownerTokenCount && currentTokenId < _currentIndex) {
      TokenOwnership memory ownership = _ownerships[currentTokenId];

      if (!ownership.burned) {
        if (ownership.addr != address(0)) {
          latestOwnerAddress = ownership.addr;
        }

        if (latestOwnerAddress == _owner) {
          ownedTokenIds[ownedTokenIndex] = currentTokenId;

          ownedTokenIndex++;
        }
      }

      currentTokenId++;
    }

    return ownedTokenIds;
  }

**Comment by sal-420** [comment]
I did the same. what fixed was changing the progma line to:  pragma solidity >=0.8.4 <0.9.0;
